### PR TITLE
Update the PR template

### PR DESCRIPTION
**Pull request by marcaddeo**
## Summary
This updates our default GitHub Pull Request template that we use between projects. This is the first step in getting this standardized.

As we discussed during the engineering meeting, we wanted to also include questions for browser requirements and performing a self review before submitting the PR.

@balbuf and I chatted and also thought it might be a good idea to ask if you've put a meaningful title in place and some meaningful labels.

## Metadata

| Question | Answer |
|----------|--------|
| Documentation reflects changes? | n/a
| `CHANGELOG` reflects changes? | n/a
| Unit/Functional tests reflect changes? | n/a
| Risk level | Low
| Relevant links | DOS-289

### CODE REVIEW

**Comment by johnpicozzi** [comment]
Idea - Instead of making these a yes/no or n/a (requiring copy & paste) could we use check boxes Check = yes unchecked = no or n/a. I think you can do that with markdown.

**Comment by marcaddeo** [comment]
While I like the idea, we've tried checkboxes and they sort of don't work great for the following reasons:

1. [ ] Anyone can check them
2. [ ] They create a "1 out of 5 tasks complete" type thing on the issue
3. [ ] They can't be used within a table, must be in a list

Do you have an idea of what it would look like with checkboxes? Post an example if you think you can come up with a good way to use checkboxes.

**Comment by johnpicozzi** [comment]
I was thinking more like this...

| Question | Response |
|----------|--------|
| Did you use a meaningful pull request title? | <ul><li>- [x] </li><ul> 
| Did you apply meaningful labels to the pull request? | <ul><li>- [x] </li><ul> 
| Did you perform a self review first? | <ul><li>- [x] </li><ul> 
| Documentation reflects changes? | <ul><li>- [ ] </li><ul> 
| `CHANGELOG` reflects changes? | <ul><li>- [ ] </li><ul> 
| Unit/Functional tests reflect changes? | <ul><li>- [ ] </li><ul> 
| Did you perform browser testing? | <ul><li>- [ ] </li><ul> 
| Risk level | Low, Medium, High
| Relevant links | JIRA issue, bug reports, security alerts, etc.

**Comment by marcaddeo** [comment]
Oh, that's not so bad. Guess you can use them in a table if you use HTML :o I can't seem to check yours off either, but i'm _fairly_ certain I could do that when it was in a PR summary.... And it would still show up as a task list for a PR.

Anyway, there's a slight flaw with this: you can't answer N/A now. At least, not without removing the checkbox html and stuff now.

**Comment by marcaddeo** [comment]
Anyone have any opinions on this? I kind of like the checkboxes, but I don't like that it shows up as a task list and kind of "abuses it"... Sometimes using a task list in a PR can be pretty useful when it's a long running PR.

**Comment by balbuf** [comment]
FWIW, I see to be able to click the checkboxes [here](https://github.com/oomphinc/drupal-scaffold/pull/23#issuecomment-462930433) but not [here](https://github.com/oomphinc/drupal-scaffold/pull/23#issuecomment-462936333) or [here](https://github.com/oomphinc/drupal-scaffold/pull/23#issuecomment-462937717).

I don't like the checkboxes tho for the following reasons:
- when placed in the PR body they create a progress bar, interpreting them as tasks, which is wrong (this alone is enough of a reason not to use them for this purpose)
- you have to open the PR first before having the benefit of clicking them, which kind of negates the purpose of a number of the items you are checking off; (I suppose you can switch to the preview tab and that might allow you to check them though?)
- idk if this can be changed with permissions settings, but if you (as the PR owner) can click the checkboxes to edit them, then anyone else can probably click them too (of course, the overarching issue is that anyone can edit anyone's PR bodies, but you have to go out of your way to edit the text than to click a checkbox)
- it makes it too easy to just check things off without really thinking about it (and as @marcaddeo pointed out, certain questions you should never respond "no" to, so you're not going to intentionally answer "no" for those questions, making the "yes" answer very intentional, whereas leaving a checkbox unchecked is a lot easier to be done "unintentionally")

**Comment by balbuf** [comment]
perhaps it would speed things up if we left each box blank and simply expected you to enter `Y`, `N`, or `N/A` (for the yes/no questions) - I think that's a bit easier than selecting and deleting the parts you don't want

**Comment by balbuf** [comment]
but considering we don't supply `N/A` as an option, I guess that isn't exactly an issue (I was imagining `yes/no/N/A` and having to delete both sides to choose "no")

**Comment by marcaddeo** [comment]
Yeah those are definitely good points against the checkboxes @balbuf, especially the fact that you have to open the PR _first_. Kind of defeats the purpose of the questions at that point.

**Comment by marcaddeo** [comment]
@balbuf so do you think it should be `yes/no/N/A` is what you're saying?

**Comment by johnpicozzi** [comment]
Maybe something where we bold the response. Like Y | **N** | N/A

**Comment by balbuf** [comment]
@marcaddeo no I _thought_ it was `yes/no/N/A` and you had to select and delete both `yes/` and `/N/A`, but you don't - you only ever have to select one half or the other and hit backspace or select the whole thing and replace with `N/A`

however, leaving it blank and specifying at the end of the question (Y/N) might be good - I think it would make it _marginally_ faster to fill out since you're not selecting/deleting anything

**Comment by balbuf** [comment]
it's also pretty obvious the questions that are asking for a yes/no answer lol esp when the template is commonplace

**Comment by balbuf** [comment]
@johnpicozzi I think bolding makes it harder to fill out and more confusing to read, plus I'm not sure how it would read out to Brandon

**Comment by nathandentzau** [comment]
giphy

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,14 +1,23 @@
-<!-- Please use a meaningful pull request title -->
-<!-- Please apply meaningful GitHub Labels to your pull request such as: PHP, Twig, SCSS, JS, etc. -->
+<!-- INSTRUCTIONS
+- Please use a meaningful pull request title which does not include the issue key or branch name.
+- Please apply meaningful GitHub Labels to your pull request such as: PHP, Twig, SCSS, JS, etc.
+- Please make sure you've reviewed the "Files changed" tab before opening this PR to ensure it includes what you expect (and nothing more) and adheres to our coding standards.
```
Comment (balbuf) [inline .github/pull_request_template.md]: I'm surprised _I'm_ the one saying this to _you_, but I think this long line should be wrapped @marcaddeo  😝 

```
- Please make sure you've reviewed the "Files changed" tab before opening this PR to ensure it
  includes what you expect (and nothing more) and adheres to our coding standards.
```

Comment (marcaddeo) [reply .github/pull_request_template.md]: Ha. Yeah it should. I didn't wrap it before because we had 1 comment per line.

**Pull request merged**